### PR TITLE
feat/weeklyview drag drop

### PR DESCRIPTION
**Pull request by KryptKode**
**Notes**
- add drag and drop feature in weekly view
- should address [this](https://github.com/SimpleMobileTools/Simple-Calendar/issues/165)

Screenshot

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/fragments/WeekFragment.kt [hunk 1]
```diff
@@ -519,6 +561,20 @@ class WeekFragment : Fragment(), WeeklyCalendar {
                                 startActivity(this)
                             }
                         }
+
+                        setOnLongClickListener { view ->
+                            currentlyDraggedView = view
+                            val shadowBuilder = View.DragShadowBuilder(view)
+                            val clipData = ClipData.newPlainText("event_id_label", event.id.toString())
+                            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/fragments/WeekFragment.kt:569]: use isNougatPlus instead please

**Comment by tibbi** [comment]
lets preserve event duration please, it acts weird now. If an event is from 17:00 to 17:00 and I pull it, it becomes an 1 hour long event. If an event is 30 minutes long by default and I drag it, it becomes a 90 minutes long event.

**Comment by KryptKode** [comment]
> lets preserve event duration please, it acts weird now. If an event is from 17:00 to 17:00 and I pull it, it becomes an 1 hour long event. If an event is 30 minutes long by default and I drag it, it becomes a 90 minutes long event.

So the dragging only affects the start time? What should happen when it is dragged to the all-day part?

**Comment by tibbi** [comment]
yes, it affects only the start time. Let's not allow dragging to the all-day part yet, lets just release some initial version for now. Also if you release the event somewhere, it blinks for a moment at the old position. Would be good to have it removed too.

**Comment by KryptKode** [comment]
@tibbi  Please check

**Comment by tibbi** [comment]
if I drag an event and release it at the same time, it disappears

**Comment by tibbi** [comment]
it disappears also if you drag it at the All-day section or the bottom week length slider

**Comment by tibbi** [comment]
yeah, looks good now. Thanks

**Pull request merged**